iroh-api/src/api_ext.rs

```diff
@@ -29,6 +29,16 @@ pub trait ApiExt: Api {
         save_get_stream(&root_path, blocks).await?;
         Ok(root_path)
     }
+
+    async fn add(&self, path: &Path, wrap: bool) -> Result<Cid> {
```

Comment:
So the code to figure out if it is a file or a dir is moved to ApiExt, and the main fns in api require you to be explicit? I like it.

Reply:
Yes, and the check for the `recursive` flag before copying a directory was removed from the API entirely, because the use case for writing code like this is very different from the reason it's in the CLI (which is to prevent accidents in interactive mode).